i18n-helpers/src/bin/mdbook-xgettext.rs

```diff
@@ -86,21 +103,161 @@ fn create_catalog(ctx: &RenderContext) -> anyhow::Result<Catalog> {
         add_message(&mut catalog, line, &source);
     }
 
+    let mut catalogs = HashMap::new();
+    // The catalog from the summary data will exist in the single pot file for
+    // a depth of 0, will exist in a top-level separate `summary.pot` file for
+    // a depth of 1, or exist within in a `summary.pot` file within the default
+    // directory for chapters without a corresponding part title.
+    let mut current_top_level = "summary".to_owned();
+    let summary_template_destination = if depth == 0 {
```

Comment:
This is perfectly fine, but in case you didn't know, you could use a `match` here:

```rust
match depth {
  0 => ...
  1 => ...
  _ => ...
}
```

Reply:
I like it! Added